packages/block-library/src/media-text/media-container.native.js

```diff
@@ -55,107 +89,172 @@ class MediaContainer extends Component {
 	}
 
-	calculateSize() {
-		if ( this.media === undefined || this.container === undefined ) {
-			return;
-		}
-
-		const { width, height } = calculatePreferedImageSize( this.media, this.container );
-		this.setState( { width, height } );
+	finishMediaUploadWithFailure() {
+		this.setState( { isUploadInProgress: false } );
 	}
 
-	onLayout( event ) {
-		const { width, height } = event.nativeEvent.layout;
-		this.container = {
-			clientWidth: width,
-			clientHeight: height,
-		};
-		this.calculateSize();
+	mediaUploadStateReset() {
+		const { onMediaUpdate } = this.props;
+
+		onMediaUpdate( { id: null, url: null } );
+		this.setState( { isUploadInProgress: false } );
 	}
 
-	renderImage() {
-		const { mediaAlt, mediaUrl } = this.props;
+	renderImage( params, openMediaOptions ) {
+		const { isUploadInProgress } = this.state;
+		const { mediaAlt, mediaUrl, isSelected } = this.props;
+		const { finalWidth, finalHeight, imageWidthWithinContainer, isUploadFailed, retryMessage } = params;
+		const opacity = isUploadInProgress ? 0.3 : 1;
 
 		return (
-			<View onLayout={ this.onLayout }>
-				<ImageBackground
-					accessible={ true }
-					//disabled={ ! isSelected }
-					accessibilityLabel={ mediaAlt }
-					accessibilityHint={ __( 'Double tap and hold to edit' ) }
-					accessibilityRole={ 'imagebutton' }
-					style={ { width: this.state.width, height: this.state.height } }
-					resizeMethod="scale"
-					source={ { uri: mediaUrl } }
-					key={ mediaUrl }
-				>
-				</ImageBackground>
-			</View>
+			<TouchableWithoutFeedback
+				accessible={ ! isSelected }
+				onPress={ this.onMediaPressed }
+				onLongPress={ openMediaOptions }
+				disabled={ ! isSelected }
+			>
+				<View style={ styles.content } >
+					{ ! imageWidthWithinContainer &&
+						<View style={ styles.imageContainer } >
+							{ this.getIcon( false ) }
+						</View> }
+					<ImageBackground
+						accessible={ true }
+						accessibilityLabel={ mediaAlt }
+						accessibilityHint={ __( 'Double tap and hold to edit' ) }
+						accessibilityRole={ 'imagebutton' }
+						style={ { width: finalWidth, height: finalHeight, opacity } }
+						resizeMethod="scale"
+						source={ { uri: mediaUrl } }
+						key={ mediaUrl }
+					>
+						{ isUploadFailed &&
+							<View style={ [ styles.imageContainer, styles.uploadFailed ] } >
+								<View style={ styles.modalIcon }>
+									{ this.getIcon( isUploadFailed ) }
+								</View>
+								<Text style={ styles.uploadFailedText }>{ retryMessage }</Text>
+							</View>
+						}
+					</ImageBackground>
+				</View>
+			</TouchableWithoutFeedback>
 		);
 	}
 
-	renderVideo() {
-		const style = { videoContainer: {} };
+	renderVideo( params, openMediaOptions ) {
+		const { mediaUrl, isSelected } = this.props;
+		const { isUploadInProgress } = this.state;
+		const { isUploadFailed, retryMessage } = params;
+		const videoHeight = ( Dimensions.get( 'window' ).width / 2 ) / VIDEO_ASPECT_RATIO;
```

Comment:
I think you can set the [aspectRatio](https://facebook.github.io/react-native/docs/layout-props#aspectratio) style prop directly and avoid the calculation. Otherwise, I think using the window width would be problematic because the ratio should apply to the container width, not the full window, but I haven't been able to test this because of the other issue I commented

Reply:
Confirmed that this gets the wrong aspect ratio when uploading a video on iPad:

![IMG_D8955C916037-1](https://user-images.githubusercontent.com/8739/65592632-39875a80-df8f-11e9-9c38-03102de1e594.jpeg)

Reply:
Nice! I'll check `aspectRatio` and also make sure it looks great on tablets as well